c++/src/kj/common-test.c++

```diff
@@ -1039,6 +1040,12 @@ KJ_TEST("single item arrayPtr()") {
   KJ_EXPECT(obj.d == 0);
 }
 
+KJ_TEST("ArrayPtr:split") {
+  auto buffer = "abc def ghij"_kjb;
+  auto buffers = buffer.split(' ');
+  KJ_EXPECT(buffers.size() == 3);
+}
```

Comment:
In some of the practical use cases we have (for instance, see my new workerd PR  implementing the `data:` URL support ... we might actually need to split on the last instance of delim, or perhaps split *only* around the first instance. My fear here is that, `split(...)` as defined here will only be useful for a limited number of cases

Reply:
I see people reaching out to std:: streams in our code base to do splits using `\n` and converting things back. So it is definitely usefull.

Happy to rename. `splitAll()`?

Reply:
Oh, no doubt it'll be useful, I just think we need to be considering the other use cases here too.

Another api alternative here could be to have `split(...)` return a bidirectional iterator rather than a `kj::Array` so that if I want to limit the number of splits, or reverse the iterator to start from the back, it would be trivial to do so

Reply:
You just use `kj::Array` as bi-di iterator, can you? It would be a bit vasteful, but at the very least the code pattern would already be correct and future-proof.

> Obviously, tho, that's a more complicated implementation so consider it a non-blocking suggestion

Sure, I'm game for a follow up iterator implementation.